core/src/main/scala/org/broadinstitute/dsp/workbench/welder/BackgroundTask.scala

```diff
@@ -106,7 +106,24 @@ class BackgroundTask(
     }
   }
 
-  def safeDelocalize(gsPath: GsPath, localObjectPath: RelativePath)(implicit ev: Ask[IO, TraceId]): IO[Unit] =
+  def checkSyncStatus(gsPath: GsPath, localObjectPath: RelativePath)(implicit ev: Ask[IO, TraceId]): IO[Unit] =
+    for {
+      traceId <- ev.ask[TraceId]
+      hashedOwnerEmail <- IO.fromEither(hashString(config.ownerEmail.value))
+      bucketMetadata <- googleStorageAlg.retrieveUserDefinedMetadata(gsPath, traceId)
+      _ <- bucketMetadata match {
+        case Some(meta) => {
+          if (meta.keySet.contains(hashedOwnerEmail.asString) && meta.get(hashedOwnerEmail.asString).getOrElse(None) == "doNotSync") {
```

Comment:
`None` is not a `String`....`meta.get(hashedOwnerEmail.asString).getOrElse(None) == "doNotSync"` is comparing `None` and `doNotSync`, which will always be `false`...it happens to work fine in this case...but this is not great.

`meta.get(hashedOwnerEmail.asString).contains("doNotSync")` will be better...also some comments about where this `doNotSync` comes from will be useful